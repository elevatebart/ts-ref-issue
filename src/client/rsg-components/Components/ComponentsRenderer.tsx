import React from 'react';
import PropTypes from 'prop-types';

export default function ComponentsRenderer({ children }: { children: React.ReactNode }) {
	return <div>{children}</div>;
}
ComponentsRenderer.propTypes = {
	children: PropTypes.node.isRequired,
};
