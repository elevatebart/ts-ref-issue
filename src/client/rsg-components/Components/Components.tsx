import React from 'react';
import PropTypes from 'prop-types';
import ReactComponent from 'rsg-components/ReactComponent';
import ComponentsRenderer from 'rsg-components/Components/ComponentsRenderer';

interface ComponentsProps {
	components: any[];
	depth: number;
	exampleMode?: string;
	usageMode?: string;
}

const Components: React.FunctionComponent<ComponentsProps> = ({
	components,
	depth,
	exampleMode,
	usageMode,
}) => {
	return (
		<ComponentsRenderer>
			{components.map(component => (
				<ReactComponent
					key={component.filepath}
					component={component}
					exampleMode={exampleMode}
					usageMode={usageMode}
					depth={depth}
				/>
			))}
		</ComponentsRenderer>
	);
};

Components.propTypes = {
	components: PropTypes.array.isRequired,
	depth: PropTypes.number.isRequired,
	exampleMode: PropTypes.string.isRequired,
	usageMode: PropTypes.string.isRequired,
};

export default Components;
