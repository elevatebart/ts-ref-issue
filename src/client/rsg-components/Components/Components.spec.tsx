import React from 'react';
import { shallow } from 'enzyme';
import ReactComponent from '../ReactComponent';
import Components from './Components';
import ComponentsRenderer from './ComponentsRenderer';
import { ExampleModes, UsageModes } from '../../consts';

const exampleMode = ExampleModes.collapse;
const usageMode = UsageModes.collapse;
const components = [
	{
		name: 'Foo',
		pathLine: 'components/foo.js',
		filepath: 'components/foo.js',
		props: {
			description: 'Foo foo',
		},
	},
	{
		name: 'Bar',
		pathLine: 'components/bar.js',
		filepath: 'components/bar.js',
		props: {
			description: 'Bar bar',
		},
	},
];

it('should render components list', () => {
	const actual = shallow(
		<Components components={components} exampleMode={exampleMode} usageMode={usageMode} depth={3} />
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render components list layout', () => {
	const actual = shallow(
		<ComponentsRenderer>
			<ReactComponent
				key={0}
				component={components[0]}
				exampleMode={exampleMode}
				usageMode={usageMode}
				depth={3}
			/>
			<ReactComponent
				key={1}
				component={components[1]}
				exampleMode={exampleMode}
				usageMode={usageMode}
				depth={3}
			/>
		</ComponentsRenderer>
	);

	expect(actual).toMatchSnapshot();
});
