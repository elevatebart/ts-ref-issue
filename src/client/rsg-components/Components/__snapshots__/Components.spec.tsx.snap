// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render components list layout 1`] = `
<div>
  <ReactComponent
    component={
      Object {
        "filepath": "components/foo.js",
        "name": "Foo",
        "pathLine": "components/foo.js",
        "props": Object {
          "description": "Foo foo",
        },
      }
    }
    depth={3}
    exampleMode="collapse"
    key="0"
    usageMode="collapse"
  />
  <ReactComponent
    component={
      Object {
        "filepath": "components/bar.js",
        "name": "Bar",
        "pathLine": "components/bar.js",
        "props": Object {
          "description": "Bar bar",
        },
      }
    }
    depth={3}
    exampleMode="collapse"
    key="1"
    usageMode="collapse"
  />
</div>
`;

exports[`should render components list 1`] = `
<ComponentsRenderer>
  <ReactComponent
    component={
      Object {
        "filepath": "components/foo.js",
        "name": "Foo",
        "pathLine": "components/foo.js",
        "props": Object {
          "description": "Foo foo",
        },
      }
    }
    depth={3}
    exampleMode="collapse"
    key="components/foo.js"
    usageMode="collapse"
  />
  <ReactComponent
    component={
      Object {
        "filepath": "components/bar.js",
        "name": "Bar",
        "pathLine": "components/bar.js",
        "props": Object {
          "description": "Bar bar",
        },
      }
    }
    depth={3}
    exampleMode="collapse"
    key="components/bar.js"
    usageMode="collapse"
  />
</ComponentsRenderer>
`;
